tt_eager/tensor/tensor_impl.hpp

```diff
@@ -417,14 +417,14 @@ DeviceBuffer allocate_buffer_on_device(
 
 template <typename T>
 std::vector<T> read_data_from_device(const Tensor &tensor, uint32_t size_in_bytes) {
```

Comment:
Is there any way to read device data with pre-allocated host memory space given as a void pointer?
Cos this API internally allocate/deallocate host memory space for device data read.
(+ I would like to remove that std::vector type)

Reply:
What API should I use to read the data of device buffer in ttDNN level?
Is this one?

Reply:
I *think* `Tensor` has some way to "steal" pre-allocated memory (say from a numpy tensor), @arakhmati can you confirm? I haven't used much of ttnn/eager myself yet

Reply:
see https://github.com/tenstorrent-metal/tt-metal/blob/22c2c6299a3c59ca59fccedfca7d36adaa07b7b6/tt_eager/tensor/borrowed_buffer.hpp#L4

Reply:
Sorry, just saw this. Yes, we can borrow memory from numpy and torch we can add support for other buffers

Reply:
And basically what I suggest is creating a tensor on host that borrows memory from some buffer. And then calling memcpy function I added to copy the data from the device tensor to the host tensor